### ISSUE TITLE
Registry operator host alias for registry viewer not working

### ISSUE THREAD

**Issue by michael-valdron** [devfile/api#1028]
## Which area this feature is related to?

/kind bug

### Which area this bug is related to?

/area registry

### Go project

**Operating System and version:** N/A

**Go Pkg Version:** 1.18

## Bug Summary

**Describe the bug:**

For the registry operator, the host alias should be able to be set after the operator is deployed and during the deployment of the devfile registry using the `(*registryv1alpha1.DevfileRegistry).Status.URL` [devfile/registry-operator/pkg/registry/configmap.go#L56](https://github.com/devfile/registry-operator/blob/9de0213a009450798a8e67b3041341a95122d123/pkg/registry/configmap.go#L56) which is set once the operator picks up the server hostname. However, in the current state `cr.Status.URL` is an empty string resulting in the registry viewer defaulting to the staging host alias of the community registry. A fix to operator will be needed to correct this.

**To Reproduce:**

Deploy the registry operator in any deployment environment, this will result in all stack/sample links (e.g. starter project download and raw devfile links) in the registry viewer pointing to the registry server to point to the one deployed in staging for the community registry.

## Expected behavior

Stack/sample links in the registry viewer should point to the registry server deployed with it. `cr.Status.URL` needs to be set to the host address of the deployment clusters ingress route.

### Any workaround?

This can also work with the following source, but this would require the registry admin to set the hostname before deploying the operator:

```go
viewerEnvfile := fmt.Sprintf(`
ANALYTICS_WRITE_KEY=%s
DEVFILE_REGISTRIES=[{"name":"Community","url":"http://localhost:8080","fqdn":"http://%s.%s"}]`,
		cr.Spec.Telemetry.RegistryViewerWriteKey, IngressName(cr.Name), cr.Spec.K8s.IngressDomain)
```

`"http://%s.%s"`: Additional source will be needed to set the right protocol based on `spec.tls.enabled`.

### Suggestion on how to fix the bug

A proper fix would have the operator to set `cr.Status.URL` to the host address of the deployment cluster's ingress route.

Target Date:  May 30, 2023

**Comment by michael-valdron** [comment]
Might be resolved by devfile/registry-operator#33

**Comment by michael-valdron** [comment]
> Might be resolved by [devfile/registry-operator#33](https://github.com/devfile/registry-operator/pull/33)

As of writing this issue has not been resolved, will need further investigating into why `cr.Status.URL` is blank on deployment.

**Comment by kim-tsao** [comment]
> > Might be resolved by [devfile/registry-operator#33](https://github.com/devfile/registry-operator/pull/33)
> 
> As of writing this issue has not been resolved, will need further investigating into why `cr.Status.URL` is blank on deployment.

I noticed this too.   I think I had a patch with the fix when I was testing at one point and then it got removed when I committed the fix.  Looking back at some of my local changes, I see there was a call to update the config map (after the status update) in the `devfileregistry_controller.go` file.  I thought I created a patch for this.  Let me look around


		//update the config map
		r.ensure(ctx, devfileRegistry, &corev1.ConfigMap{}, labels, "")
		if result != nil {
			return *result, err
		}

**Comment by michael-valdron** [comment]
This bug seems to have resurfaced again and requires further investigation, reopening..

**Comment by michael-valdron** [comment]
Added this issue to the current sprint plan, this bug was fixed in sprint 233 but resurfaced at some point during the last sprint. Further investigation into this is require to ensure a fully working registry viewer component under a devfile registry deployment via the registry operator.

**Comment by kim-tsao** [comment]
Another bug to fix as part of epic https://github.com/devfile/api/issues/1007

**Comment by kim-tsao** [comment]
It looks like the configmap is being updated but not the deployment

**Comment by michael-valdron** [comment]
> Another bug to fix as part of epic #1007

This epic is now being tracked by #410.
